workshops/speak_colors/README.md

```diff
@@ -0,0 +1,202 @@
+- We’re linking the two external libraries we need
+- Finally, a link to the `script.js` file we’ll write our own JavaScript code in
+
+The rest of the code samples in this workshop will all be for `script.js`, where we write the p5 code.
+
+## Displaying instructions
+
+First, we need to set up a canvas with p5.
+
+- We want the canvas to fill the screen, so JavaScript’s `window.innerWidth` & `window.innerHeight` come in handy.
+- Let’s go ahead & fill the background with white, and make the text color dark grey.
+
+```js
+function setup() {
+  createCanvas(window.innerWidth, window.innerHeight)
+  background(255)
+  fill(25)
+}
+```
+
+The page still looks empty, oops! Let’s add some initial instructions, even though they won’t do anything yet.
+
+(You can use only the last line, which calls `text()`, but the other lines make the text big, chunky, & more fun-looking.)
+
+```js
+function setup() {
+  createCanvas(window.innerWidth, window.innerHeight)
+  background(255)
+  fill(25)
+
+  textSize(48)
+  textAlign(CENTER)
+  textStyle(BOLDITALIC)
+  textFont('"Avenir Next", system-ui, sans-serif')
+  text('SAY A COLOR', width / 2, height / 2)
+}
+```
+
+Fantastic! Up next: the speech recognition we came here for.
+
+## Running speech recognition
+
+The p5.speech library has [an example](https://github.com/IDMNYU/p5.js-speech/blob/master/examples/05continuousrecognition.html) of "continuous" speech recognition--that is, the mic stays active after you’ve said your first phrase, instead of being one-time.
+
+Let’s set up continuous speech recognition, and display an alert when new speech is transcribed:
+
+```js
+const speech = new p5.SpeechRec('en-US', parseResult)
+speech.continuous = true
+speech.interimResults = false
+
+function setup() {
+  // …
+  text('SAY A COLOR', width / 2, height / 2)
+  speech.start()
+}
+
+function draw() {
+  // Where we’re going we don’t need drawing
+}
+
+function parseResult() {
+  if (speech.resultValue) {
+    alert(speech.resultString)
+  }
+}
+```
+
+We’re getting there…
+
+## Changing the color
+
+Because HTML/CSS support a wide array of [named colors](https://html-color-codes.info/color-names/), we’re going to use those for this project. (You can read more about [where the color names come from](https://www.chenhuijing.com/blog/where-did-css-named-colours-come-from/), or if you want to extend this project, figure out how to use [JavaScript objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object) to support your own colors!)
+
+For the first iteration, we’re just supporting single-word colors, because it’s simplest.
+
+- We’re starting with the `speech.resultString`, which we used before.
+- To get the last word, we’ll do `.split(' ')`, which converts the text into a [JavaScript array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array), then `.pop()`, which gets the last item from the array. This is a string.
```

Comment:
imo it would be helpful to elaborate on this a little bit more.
Yes, it makes an array, but not just an array, an array of words.
It decides what a "word" is by grabbing all of the text between individual spaces, that's why we pass `' '` to `.split()`.

Reply:
It also might be helpful to link to the split docs. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split

Reply:
Addressed this, thanks!